nemo/collections/tts/models/ssl_tts.py

```diff
@@ -0,0 +1,524 @@
+                self.stop_gradient = self._cfg.get('stop_gradient', False)
+                assert (
+                    self.stop_gradient and self.augment_ctc
+                ) == False, "stop_gradient and augment_ctc cannot be true at the same time"
+                self.mse_loss = nn.MSELoss()
+
+                self.ctc_decoder = GreedyCTCDecoder(self._text_tokenizer.tokens, self._text_tokenizer.blank)
+
+        self.automatic_optimization = False
+
+        stft_cfg = self._cfg.preprocessor
+        librosa_mel_filter = librosa.filters.mel(
+            sr=stft_cfg.sample_rate, n_fft=stft_cfg.n_fft, n_mels=stft_cfg.features, fmin=0, fmax=8000
+        )
+        fb = torch.tensor(librosa_mel_filter, dtype=torch.float,).unsqueeze(0)
+
+        self.register_buffer("fb", fb)
+
+    @classmethod
+    def list_available_models(cls) -> Optional[PretrainedModelInfo]:
+        """
+        This method returns a list of pre-trained model which can be instantiated directly from NVIDIA's NGC cloud.
+
+        Returns:
+            List of available pre-trained models.
+        """
+        results = []
+
+        model = PretrainedModelInfo(
+            pretrained_model_name="ssl_en_conformer_large",
+            description="For details about this model, please visit https://ngc.nvidia.com/catalog/models/nvidia:nemo:ssl_en_conformer_large",
+            location="https://api.ngc.nvidia.com/v2/models/nvidia/nemo/ssl_en_conformer_large/versions/1.10.1/files/ssl_en_conformer_large.nemo",
+        )
+        results.append(model)
+
+        model = PretrainedModelInfo(
+            pretrained_model_name="ssl_en_conformer_xlarge",
+            description="For details about this model, please visit https://ngc.nvidia.com/catalog/models/nvidia:nemo:ssl_en_conformer_xlarge",
+            location="https://api.ngc.nvidia.com/v2/models/nvidia/nemo/ssl_en_conformer_xlarge/versions/1.10.0/files/ssl_en_conformer_xlarge.nemo",
+        )
+        results.append(model)
+
+        return results
+
+    @property
+    def tb_logger(self):
+        if self._tb_logger is None:
+            if self.logger is None and self.logger.experiment is None:
+                return None
+            tb_logger = self.logger.experiment
+            if isinstance(self.logger, LoggerCollection):
+                for logger in self.logger:
+                    if isinstance(logger, TensorBoardLogger):
+                        tb_logger = logger.experiment
+                        break
+            self._tb_logger = tb_logger
+        return self._tb_logger
+
+    def __setup_dataloader_from_config(self, data_config):
+        # _text_tokenizer = instantiate(self._cfg.text_tokenizer)
+        if hasattr(self, '_text_tokenizer') and isinstance(self._text_tokenizer, BaseTokenizer):
+            _text_tokenizer = self._text_tokenizer
+        else:
+            _text_tokenizer = self._text_tokenizer = EnglishCharsTokenizer(add_blank_at="last")
```

Comment:
Should this warn before overriding if `_test_tokenizer` is set but not a `BaseTokenizer`?

Reply:
added warning